src/data/reports/glass-admin/GLASSDataMaintenanceDefaultRepository.ts

```diff
@@ -0,0 +1,492 @@
+
+                await this.globalStorageClient.saveObject<ATCJson>(`ATC-${year}-v1`, jsons);
+                await this.globalStorageClient.saveObject<ATCJson>(Namespaces.ATCS, updatedATCItems);
+            }
+        } else {
+            throw new Error("The zip file does not contain exactly 4 JSON files.");
+        }
+    }
+
+    async getRecalculationLogic(namespace: string): Promise<AMCRecalculation | undefined> {
+        return await this.globalStorageClient.getObject<AMCRecalculation>(namespace);
+    }
+
+    async cancelRecalculation(namespace: string): Promise<void> {
+        const amcRecalculationLogic = await this.getRecalculationLogic(namespace);
+
+        if (amcRecalculationLogic) {
+            const updatedRecalculationLogic = {
+                ...amcRecalculationLogic,
+                recalculate: false,
+            };
+
+            await this.globalStorageClient.saveObject<AMCRecalculation>(namespace, updatedRecalculationLogic);
+        }
+    }
+
+    async saveRecalculationLogic(namespace: string, atcNamespace: string): Promise<void> {
+        const amcRecalculationLogic = await this.getRecalculationLogic(namespace);
+        const atcs = await this.getATCItems("ATCs");
+
+        const updatedATCs: ATCItem[] = atcs.map(atc => {
+            if (atc.currentVersion) {
+                return {
+                    ...atc,
+                    previousVersion: true,
+                };
+            } else {
+                return {
+                    ...atc,
+                    previousVersion: false,
+                };
+            }
+        });
+
+        const currentDate = new Date().toISOString();
+        const periods = this.getATCPeriods();
+        const enrolledCountries = await this.getEnrolledCountries();
+
+        const amcRecalculation: AMCRecalculation = {
+            date: currentDate,
+            periods: periods,
+            orgUnitsIds: enrolledCountries,
+            loggerProgram: amcRecalculationLogic?.loggerProgram ?? "",
+            recalculate: true,
+        };
+
+        this.globalStorageClient.saveObject<AMCRecalculation>(namespace, amcRecalculation);
+        this.globalStorageClient.saveObject<ATCItem[]>(atcNamespace, updatedATCs);
+    }
+
+    async getLoggerProgramName(programId: string): Promise<string> {
+        const { programs } = await this.api.metadata
+            .get({
+                programs: {
+                    fields: {
+                        id: true,
+                        name: true,
+                    },
+                    filter: {
+                        id: {
+                            eq: programId,
+                        },
+                    },
+                },
+            })
+            .getData();
+
+        return programs[0]?.name ?? "";
+    }
+
+    private getATCPeriods = (): string[] => {
+        const currentYear = new Date().getFullYear();
+        const startYear = 2016;
```

Comment:
Maybe it's better to create a constant in the upper part of the code named like `const START_YEAR_PERIODS = 2016;` to make it easier to locate and better understand that it is the first year used in the project to start the periods.